tests/contrib/langchain/test_langchain.py

```diff
@@ -0,0 +1,681 @@
+        file="lcel_openai_chat_streamed_response.txt",
+    )
+    model = langchain_openai.ChatOpenAI(client=client)
+
+    for _ in model.stream(input="how can langsmith help with testing?"):
+        pass
+
+
+@pytest.mark.snapshot(ignores=IGNORE_FIELDS)
+def test_streamed_llm(langchain_openai, streamed_response_responder):
+    client = streamed_response_responder(
+        module="openai",
+        client_class_key="OpenAI",
+        http_client_key="http_client",
+        endpoint_path=["completions"],
+        file="lcel_openai_llm_streamed_response.txt",
+    )
+
+    llm = langchain_openai.OpenAI(client=client)
+
+    for _ in llm.stream(input="How do I write technical documentation?"):
+        pass
+
+
+@pytest.mark.snapshot(
+    ignores=IGNORE_FIELDS,
+    token="tests.contrib.langchain.test_langchain.test_streamed_chain",
+)
+async def test_astreamed_chain(langchain_core, langchain_openai, async_streamed_response_responder):
+    client = async_streamed_response_responder(
+        module="openai",
+        client_class_key="AsyncOpenAI",
+        http_client_key="http_client",
+        endpoint_path=["chat", "completions"],
+        file="lcel_openai_chat_streamed_response.txt",
+    )
+
+    prompt = langchain_core.prompts.ChatPromptTemplate.from_messages(
+        [("system", "You are world class technical documentation writer."), ("user", "{input}")]
+    )
+    llm = langchain_openai.ChatOpenAI(async_client=client)
+    parser = langchain_core.output_parsers.StrOutputParser()
+
+    chain = prompt | llm | parser
+    async for _ in chain.astream({"input": "how can langsmith help with testing?"}):
+        pass
+
+
+@pytest.mark.snapshot(
+    ignores=IGNORE_FIELDS,
+    token="tests.contrib.langchain.test_langchain.test_streamed_chat",
+)
+async def test_astreamed_chat(langchain_openai, async_streamed_response_responder):
+    client = async_streamed_response_responder(
+        module="openai",
+        client_class_key="AsyncOpenAI",
+        http_client_key="http_client",
+        endpoint_path=["chat", "completions"],
+        file="lcel_openai_chat_streamed_response.txt",
+    )
+
+    model = langchain_openai.ChatOpenAI(async_client=client)
+
+    async for _ in model.astream(input="how can langsmith help with testing?"):
+        pass
+
+
+@pytest.mark.snapshot(
+    ignores=IGNORE_FIELDS,
+    token="tests.contrib.langchain.test_langchain.test_streamed_llm",
+)
+async def test_astreamed_llm(langchain_openai, async_streamed_response_responder):
+    client = async_streamed_response_responder(
+        module="openai",
+        client_class_key="AsyncOpenAI",
+        http_client_key="http_client",
+        endpoint_path=["completions"],
+        file="lcel_openai_llm_streamed_response.txt",
+    )
+
+    llm = langchain_openai.OpenAI(async_client=client)
+
+    async for _ in llm.astream(input="How do I write technical documentation?"):
+        pass
+
+
+# TODO: needs fixing in follow-up
```

Comment:
note: I'm not too sure what's going on with this test, it only ever failed on small snapshot asserts, but I didn't want to block more time on it for now. I will follow up with it after we get in 3.0 changes

Reply:
Let's add this as a follow up JIRA item!

Reply:
Will add it as a note to the existing ticket for langchain flakiness!